src/community/json-ld/src/main/java/org/geoserver/jsonld/builders/impl/DynamicValueBuilder.java

```diff
@@ -141,4 +144,19 @@ private String workXpathFunction(String expression) {
         }
         return expression;
     }
+
+    private boolean canWriteValue(Object result) {
```

Comment:
I would add a comment describing briefly what the method does and what the invoker should be aware of.

Reply:
Suggestion:
```
/**
     * A value can only be wrote if it is non NULL and not an empty list. This method supports
     * complex features attributes, this method will be invoked recursively on the attribute value.
     */
    private boolean canWriteValue(Object result) {
        if (result instanceof Attribute) {
            // this is an attribute from a complex feature let's test its value
            return canWriteValue(((Attribute) result).getValue());
        } else if (result instanceof List) {
            // the value is a list, let's check it is not empty
            return ((List) result).size() != 0;
        }
        // let's check we don't have a NULL value
        return result != null;
    }
```

Reply:
fixed thanks